src/libs/ajax/workspaces/providers/WorkspaceProvider.ts

```diff
@@ -0,0 +1,22 @@
+import { Ajax } from 'src/libs/ajax';
+import { AbortOption } from 'src/libs/ajax/data-provider-common';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export type FieldsArg = string[];
+export type WorkspaceListOptions = AbortOption & {
+  stringAttributeMaxLength?: string | number;
+};
+
+// TODO: grow this interface to include more of the available data verbs
+export interface WorkspaceDataProvider {
+  list: (fieldsArgs: FieldsArg, options: WorkspaceListOptions) => Promise<WorkspaceWrapper[]>;
+}
+
+export const workspaceProvider: WorkspaceDataProvider = {
+  list: async (fieldsArgs: FieldsArg, options: WorkspaceListOptions): Promise<WorkspaceWrapper[]> => {
+    const { signal, stringAttributeMaxLength } = options;
+
+    const ws = await Ajax(signal).Workspaces.list(fieldsArgs, stringAttributeMaxLength);
+    return ws;
```

Comment:
Style nit...

If terseness is a goal, we could return the result of `list` directly.

```suggestion
    return Ajax(signal).Workspaces.list(fieldsArgs, stringAttributeMaxLength);
```

Or if descriptiveness is a goal, then I'd suggest not abbreviating variable names.

```suggestion
    const workspaces = Ajax(signal).Workspaces.list(fieldsArgs, stringAttributeMaxLength);
    return workspaces;
```

Reply:
this was unmodified from the original code.... I will go with the 2nd suggestion.